### PR TITLE
Update compositions to allow patches from multiple inputs

### LINKED ISSUE #1: Requires patch from multiple inputs
Compositions in this repository require patching from multiple inputs. 

Tested and works with [this open PR in Crossplane.](https://github.com/crossplane/crossplane/issues/1973)

### PR DESCRIPTION
**Pull request by ezgidemirel**
Signed-off-by: ezgidemirel <ezgidemirel91@gmail.com>

### Description of your changes
This change updates compositions to allow patches from multiple inputs and work with latest Crossplane/Provider versions. 

Fixes #1 

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).

### How has this code been tested

I created a package with these configurations and applied example files to create a k8s cluster with Vault.

### CODE REVIEW

File: configuration/crossplane.yaml [hunk 1]
```diff
@@ -44,8 +44,8 @@ metadata:
 spec:
   crossplane:
     version: ">=v1.0.0-0"
```
Comment (turkenh) [inline configuration/crossplane.yaml:46]: Let's also bump minimum xp version, e.g. 1.6.0

File: configuration/cluster/gke/composition.yaml [hunk 2]
```diff
@@ -11,16 +11,13 @@ spec:
...
         spec:
           forProvider:
-            initialClusterVersion: "1.16"
+            initialClusterVersion: "1.21"
             location: us-west2
-            masterAuth:
-              # setting this master auth user name enables basic auth so that a client (e.g.,
-              # provider-helm), can connect with the generated kubeconfig from the connection secret
-              username: admin
+            serviceAccount: sample-sa
```
Comment (turkenh) [inline configuration/cluster/gke/composition.yaml:20]: Where does this sa coming? Does it needs to be created beforehand?

**Review by turkenh — COMMENTED** [review]
Looking pretty good, thanks @ezgidemirel!

Left two small comments.

File: configuration/crossplane.yaml (hunk 1 above)
Comment (ezgidemirel) [reply configuration/crossplane.yaml]: done

File: configuration/cluster/gke/composition.yaml (hunk 2 above)
Comment (ezgidemirel) [reply configuration/cluster/gke/composition.yaml]: looks like it doesn't :) removed it

**Review by turkenh — APPROVED** [review]
Great work @ezgidemirel, thanks a lot!

**Pull request merged**